py/shell.nix

```diff
@@ -0,0 +1,13 @@
+{ pkgs ? import <nixpkgs> {} } : pkgs.mkShell {
+	buildInputs = with pkgs; [
+		(python310.withPackages (pypkgs: with pypkgs; [
```

Comment:
I wonder if there's a way to get 3.11, as that ~doubled the speed compared to 3.10...

Reply:
[It's in `nixpkgs`](https://search.nixos.org/packages?channel=22.11&show=python311&from=0&size=50&sort=relevance&type=packages&query=python311); should just be a 1 character change :smile:

Reply:
Hm. On both Debian and Manjaro, switching this to Python 3.11 currently makes it take a long time as it builds from source, and then fails with missing `longintrepr.h`. I'm leaving it for now

Reply:
Not a huge deal but: @will-ca which nixpkgs commit are you on? `python311` builds fine (gets substituted from cache) on the latest commit of both the unstable and 22.11 channels (`23.05pre431956.14ddeaebcbe` and `22.11.656.52e3e80afff` respectively) for me (`x86_64`, linux)

Reply:
`"23.05pre431709.813836d64fa"` in Debian and `"23.05pre432312.543070eafe8"` in Manjaro, apparently.

Hm. `python311` works fine. It looks like it's `python311.withPackages (p: [ p.mypy ])` that takes forever, and `python311.withPackages (p: [ p.black ])` that fails with missing `longintrepr.h`.